lib/route/link/bonding.c

```diff
@@ -19,6 +19,69 @@
 #include <netlink/route/link/bonding.h>
 #include <netlink-private/route/link/api.h>
 
+/** @cond SKIP */
+#define BOND_HAS_MODE		(1 << 0)
+#define BOND_HAS_ACTIVE_SLAVE	(1 << 1)
+
+struct bond_info {
+	uint8_t bn_mode;
+	uint32_t ifindex;
+	uint32_t bn_mask;
+};
+
+/** @endcond */
+
+static int bond_alloc(struct rtnl_link *link)
+{
+	struct bond_info *bn;
+
+	if(link->l_info)
+		memset(link->l_info, 0, sizeof(*bn));
+	else {
+		bn = calloc(1, sizeof(*bn));
+		if (!bn)
+			return -NLE_NOMEM;
+
+		link->l_info = bn;
+	}
+
+	return 0;
+}
+
+static void bond_free(struct rtnl_link *link)
+{
+	struct bond_info *bn = link->l_info;
+
+	free(bn);
+	link->l_info = NULL;
+}
+
+/**
+ * Set active slave
+ * @arg link            Link object
+ * @arg active          ifindex of active slave
+ *
+ */
+void rtnl_link_set_bondactiveslave(struct rtnl_link *link, int active_slave)
+{
+        struct bond_info *bn = link->l_info;
+        bn->ifindex = active_slave;
```

Comment:
I think we should check that the link is of the expected type. See for example `rtnl_link_bridge_set_vlan_filtering()` which asserts that the type is as expected.